### ISSUE TITLE
`Image::cache` returns string if encoding happens inside the callback.

### ISSUE THREAD

**Issue by Nikita240** [Intervention/imagecache#51]
Consider the following

```php
return Image::cache(function($image) {

	$image->make('test.jpg');

	$image->encode('png');

	return $image;
}, 43200, true)->response();
```

On the first run, `Image::cache` will return a string, instead of an `Image` object like it's supposed to.

After it's cached however, `Image::cache` will return an `Image` object like it's supposed to.

**Comment by olivervogel** [comment]
Thanks for reporting. Is fixed in b3f1958c5f300a1986281d599f6a3654d991e30b.

**Comment by Nikita240** [comment]
Thanks!!!
